core/decorators.py

```diff
@@ -0,0 +1,194 @@
+from viur.core import errors, current
+from typing import Callable, Dict, Union, List
+
+import functools
+import logging
+import types
+import warnings
+
+
+def ensure_viur_flags(f: Callable) -> None:
+    try:
+        f.viur_flags
+    except AttributeError:
+        f.viur_flags = {}
+
+
+def access(*access: str | list[str]) -> Callable:
```

Comment:
Instead of raising an error it's sometimes nice to redirect the user:

```suggestion
def access(*access: str | list[str], *, open_login: boolean=False, open_login_url: str = "/user/auth_userpassword/login") -> Callable:
```

[part 1/2]